include/uapi/sound/sof/header.h

```diff
@@ -25,35 +31,21 @@
  */
 struct sof_abi_hdr {
 	__u32 magic;
-	__u32 type;
+	union {
+		struct {
+			__u32 type;
+		} ipc3;
+		struct {
+			__u8 param_id;
+			__u8 reserved[3];
+		} ipc4;
+	};
```

Comment:
we might not need this change at all?
Leaving the `type` as it were and document that it is param_id in IPC4.
Looking at the firmware code, the type is used for the same purpose as the param_id.